content/blog/2023/nextflow-with-gbatch.md

```diff
@@ -0,0 +1,288 @@
+    """
+    mkdir fastqc_${sample_id}_logs
+    fastqc -o fastqc_${sample_id}_logs -f fastq -q ${reads}
+    """
+}
+
+process MULTIQC {
+    publishDir params.outdir, mode:'copy'
+
+    input:
+    path '*'
+
+    output:
+    path 'multiqc_report.html'
+
+    script:
+    """
+    multiqc .
+    """
+}
+
+workflow {
+    Channel
+        .fromFilePairs(params.reads, checkIfExists: true)
+        .set { read_pairs_ch }
+
+    index_ch = INDEX(params.transcriptome_file)
+    quant_ch = QUANTIFICATION(index_ch, read_pairs_ch)
+    fastqc_ch = FASTQC(read_pairs_ch)
+    MULTIQC(quant_ch.mix(fastqc_ch).collect())
+}
+
+workflow.onComplete {
+    log.info ( workflow.success ? "\nDone! Open the following report in your browser --> $params.outdir/multiqc_report.html\n" : "Oops .. something went wrong" )
+}
+```
+
+#### Download the input data
+
+Create a folder named `data` in your project folder and download the example files (also taken from the Seqera training material).
+
+```bash
+$ mkdir data; cd data
+$ wget https://raw.githubusercontent.com/seqeralabs/nf-training-public/master/nf-training/data/ggal/gut_1.fq
+$ wget https://raw.githubusercontent.com/seqeralabs/nf-training-public/master/nf-training/data/ggal/gut_2.fq
+$ wget https://raw.githubusercontent.com/seqeralabs/nf-training-public/master/nf-training/data/ggal/transcriptome.fa
+$ cd ..
+```
+
+Files can also be uploaded directly to Google Cloud Storage, or downloaded directly from GitHub, depending on your own preferences.
+
+Use the command below to authenticate with Google Cloud Platform. Nextflow will use this account by default when you run a pipeline.
+
+```bash
+$ gcloud auth application-default login
+```
+
+#### Launch the pipeline!
+
+With that done, you’re now ready to run the example Nextflow pipeline. Remember the name of the bucket you created? You need to set it so that Nextflow can move our input data there and store intermediate and final data. Nextflow does not allow you to use the root directory of a bucket as the work directory -- it must be a subdirectory instead. Using a subdirectory is also just a good practice. If you remember when we wrote our `nextflow.config` file above, you probably noticed it was set there with `workDir`, but it could also be set in the command-line with the `-w` parameter. Because it is already in the `nextflow.config` file, we can launch our pipeline with:
+
+```bash
+$ nextflow run main.nf
+```
+
+While the pipeline stores everything in the bucket, our example pipeline will also download the final outputs to a local directory called `results`, because of how the `publishDir` directive was specified in the `main.nf` script. If you want to avoid the egress cost associated with downloading data from a bucket, you can change the `publishDir` to another bucket directory, e.g. `gs://rnaseq-pipeline-bckt/results`.
+
+In your terminal, you should see something like this:
+
+![Nextflow ongoing run on Google Cloud Batch](/img/ongoing-nxf-gbatch.png)
```

Comment:
Whoops, looks like this image needs to be updated to reflect the new launch command

Reply:
Good point!